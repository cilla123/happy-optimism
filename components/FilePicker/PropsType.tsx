export default interface PropsType {
  disabled?: boolean;
  multiple?: boolean;
  quality?: number;
  accept?: string;
  capture?: any;
  onChange?: (file: Object | Array<Object>) => void;
  onBeforeSelect?: () => boolean;
  children?: any;
}
