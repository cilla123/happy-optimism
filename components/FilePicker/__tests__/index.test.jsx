import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import FilePicker from '../index';

describe('FilePicker', () => {
  it('renders correctly', () => {
    const props = {
      onChange: jest.fn(),
      children: <div>add</div>,
    };
    const wrapper = render(<FilePicker {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('select file', () => {
    const props = {
      accept: 'image/jpg, image/jpeg, image/gif, image/png',
      onChange: jest.fn(),
      multiple: true,
      onBeforeSelect: jest.fn(),
      children: <div>add</div>,
    };
    const wrapper = mount(<FilePicker {...props} />);
    wrapper.find('.za-filepicker-input').simulate('change');
    // expect(props.onChange).toBeCalled();
  });
});
