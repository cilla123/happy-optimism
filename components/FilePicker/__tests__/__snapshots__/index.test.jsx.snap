// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilePicker renders correctly 1`] = `
<div
  class="za-filepicker"
>
  <input
    class="za-filepicker-input"
    type="file"
  />
  <div>
    add
  </div>
</div>
`;
